perma_web/frontend/components/CreateLinkBatch.vue

```diff
@@ -0,0 +1,216 @@
+const userSubmittedLinks = ref('')
+
+const readyStates = ["ready", "captureBatchError"]
+const isReady = computed(() => { readyStates.includes(globalStore.batchCaptureStatus) })
+
+let progressInterval;
+
+const dialogRef = ref('')
+const handleOpen = () => {
+    dialogRef.value.handleDialogOpen();
+}
+
+const handleReset = () => {
+    clearInterval(progressInterval)
+    userSubmittedLinks.value = ''
+    batchCaptureJobs.value = []
+    batchDialogTitle.value = defaultDialogTitle
+    globalStore.batchCaptureStatus = "ready"
+}
+
+const handleClose = () => {
+    handleReset()
+    dialogRef.value.handleDialogClose();
+}
+
+const handleClick = (e) => {
+    // Close dialog if users click outside of the inner dialog container
+    // For example, clicks on the dialog's ::backdrop pseudo-element will trigger this
+    if (e.target.classList.contains('c-dialog')) {
+        handleClose();
+    }
+}
+
+const handleBatchCaptureRequest = async () => {
+    if (!isReady) {
+        return
+    }
+
+    globalStore.updateBatchCaptureErrorMessage('')
+    globalStore.updateBatchCapture('isValidating')
+
+    const formData = {
+        urls: userSubmittedLinks.value.split("\n").map(s => { return s.trim() }).filter(Boolean),
+        target_folder: globalStore.selectedFolder.folderId,
+        human: true,
+    }
+
+    const csrf = getCookie("csrftoken")
+
+    try {
+        const response = await fetch("/api/v1/archives/batches/",
+            {
+                headers: {
+                    "X-CSRFToken": csrf,
+                    "Content-Type": "application/json"
+                },
+                method: "POST",
+                credentials: "same-origin",
+                body: JSON.stringify(formData)
+            })
+
+        if (!response?.ok) {
+            throw new Error(response.statusText) // This is a placeholder for now
+        }
+
+        const data = await response.json()
+        batchCaptureId.value = data // Triggers periodic polling
+        globalStore.updateBatchCapture('isQueued')
+
+        batchDialogTitle.value = "Link Batch Details"
+        batchCSVUrl.value = `/api/v1/archives/batches/${data.id}/export`
+
+        const batchCreated = new CustomEvent("BatchLinkModule.batchCreated");
+        window.dispatchEvent(batchCreated);
+
+    } catch (error) {
+        handleError(error)
+    }
+};
+
+const handleError = (error) => {
+    clearInterval(progressInterval)
+    globalStore.updateBatchCapture('urlError')
+    globalStore.updateBatchCaptureErrorMessage(error)
+    console.log(error) // This is a placeholder
+    handleClose()
+}
+
+const handleBatchDetailsFetch = async () => {
+    console.log('batch details fetch initiated')
+    const { data, error, errorMessage } = await useFetch(`/api/v1/archives/batches/${batchCaptureId.value.id}`)
+
+    if (error) {
+        return globalStore.updateBatchCaptureErrorMessage(errorMessage)
+    }
+
+    if (data.value.capture_jobs) {
```

Comment:
This is a fair question. I think this could be handled by removing the if statement here, and revising the one above it: 

```
    if (error || !data.value.capture_jobs) {
        globalStore.updateBatchCapture('batchDetailsError')
        globalStore.updateBatchCaptureErrorMessage(errorMessage)
    }
```

This is handled in [4049cc0](https://github.com/harvard-lil/perma/pull/3521/commits/4049cc013dad9cd1ce8953bdb37123ea81eec6ce)